lib/buffer.js

```diff
@@ -692,50 +692,51 @@ Buffer.prototype[customInspectSymbol] = function inspect() {
 Buffer.prototype.inspect = Buffer.prototype[customInspectSymbol];
 
 Buffer.prototype.compare = function compare(target,
-                                            start,
-                                            end,
-                                            thisStart,
-                                            thisEnd) {
+                                            targetStart,
+                                            targetEnd,
+                                            sourceStart,
+                                            sourceEnd) {
   if (!isUint8Array(target)) {
     throw new ERR_INVALID_ARG_TYPE('target', ['Buffer', 'Uint8Array'], target);
   }
   if (arguments.length === 1)
     return _compare(this, target);
 
-  if (start === undefined)
-    start = 0;
-  else if (start < 0)
-    throw new ERR_INDEX_OUT_OF_RANGE();
+  if (targetStart === undefined)
+    targetStart = 0;
+  else if (targetStart < 0)
+    throw new ERR_OUT_OF_RANGE('targetStart', '>= 0', targetStart);
   else
-    start >>>= 0;
+    targetStart >>>= 0;
 
-  if (end === undefined)
-    end = target.length;
-  else if (end > target.length)
-    throw new ERR_INDEX_OUT_OF_RANGE();
+  if (targetEnd === undefined)
+    targetEnd = target.length;
+  else if (targetEnd > target.length)
+    throw new ERR_OUT_OF_RANGE('targetEnd', undefined, targetEnd);
```

Comment:
We actually know the range and I suggest to add it to the error message:

```js
throw new ERR_OUT_OF_RANGE('targetEnd', `>= 0 and <= ${target.length}`, targetEnd)
```

Reply:
The challenge with specifying the maximum is that it is not static and I want to make sure I don't give the impression that it is.. The user will get a message that says that `targetEnd` must be <= 54 and think that `targetEnd` can never be greater than **54**, but what it really means is it can never be greater than **the length of `target`** which will be different on different calls.

I was going to add text that said `target.length` rather than `${target.length}` but the docs say `target.byteLength`. So I think there's some exploring that needs to be done about whether the supplied object was a `Buffer` or `ArrayBuffer` (one has a `.length` and the other a `.byteLength`)...

Oh, maybe it can say `">= 0 and <= the length of "target"`. I think that would be sufficient. Does that work for you?

Reply:
We currently use dynamic limits in most cases. Using the documented argument name is something tedious that would easily become outdated and it is not user friendly.

However, adding something on top of the dynamic limit would be fine for me (e.g., `>= 0 and <= ${target.length} (the "target" length)`).

Reply:
Using both works for me!

Reply:
(For that matter, just sticking with the dynamic value works for me too if it's what we're doing everywhere else. If it ever proves to be a big source of confusion for users--which the more I think about it, the less I think it will be--then we can change things globally.)

Reply:
`targetEnd` can be less than 0.

```console
> buf.compare(target, 1, -100, 1, 9)
-1
>
```

I will add the value of the maximum, though. Thanks!